Add enum `ElectricalComponentControlMode`

tiyash-basu-frequenz opened the pull request:
This enum is used to explicitly define whether an electrical component is intended to be active (enabled) or inactive (disabled). This intention directly determines the component's availability to downstream services and applications.

Note that this enum differs from the `ElectricalComponentStatus` enum, which is used to report the actual operational state of the component.

Thanks to @thomas-nicolai-frequenz for providing the original version.

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto:
```diff
@@ -121,6 +121,54 @@ enum ElectricalComponentStatus {
+//     intention to make the component either ACTIVE or INACTIVE. This directly
+//     impacts availability:
+//     - ACTIVE indicates the component is intended to be available and usable
+//       by downstream agents and services.
```
Maybe we say "agents, applications and services"?

tiyash-basu-frequenz replied:
Done

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto:
```diff
@@ -121,6 +121,54 @@ enum ElectricalComponentStatus {
+//     - ACTIVE indicates the component is intended to be available and usable
+//       by downstream agents and services.
+//     - INACTIVE indicates the component is intended to be unavailable and
+//       ignored by downstream agents and services.
```
I would say not usable instead of ignored.

tiyash-basu-frequenz replied:
I changed it to "unavailable for use", since the component should still be available, but no agent will be able to control it or use it.
Does the phrasing look okay now?

thomas-nicolai-frequenz replied:
yes.

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto:
```diff
@@ -121,6 +121,54 @@ enum ElectricalComponentStatus {
+//     damages.
+//
+// !!! example "Control Mode Change Impact"
+//     If a battery component's Control Mode is switched from ACTIVE to
```
Is that only true for batteries?

tiyash-basu-frequenz replied:
It should be true for every component that we can disable. As an example, battery works well though.

thomas-nicolai-frequenz replied:
ah its an example I missed that.

thomas-nicolai-frequenz commented on proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto:
```diff
@@ -121,6 +121,54 @@ enum ElectricalComponentStatus {
+// !!! example "Control Mode Change Impact"
+//     If a battery component's Control Mode is switched from ACTIVE to
+//     INACTIVE, it immediately becomes unavailable to downstream
+//     energy-management services, potentially disrupting scheduled
```
Lets remove the "energy-management" part and just say "services and agents".

tiyash-basu-frequenz replied:
Done

tiyash-basu-frequenz committed:
Add enum `ElectricalComponentControlMode`

This enum is used to explicitly define whether an electrical component
is intended to be active (enabled) or inactive (disabled). This
intention directly determines the component's availability to downstream
services and applications.

Note that this enum differs from the `ElectricalComponentStatus` enum,
which is used to report the actual operational state of the component.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

# Conflicts:
#	RELEASE_NOTES.md
#	proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto

tiyash-basu-frequenz committed:
Remove `ElectricalComponentStatus` enum

The `ElectricalComponentStatus` enum has been removed in favour of the
`ElectricalComponentControlMode` enum.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

# Conflicts:
#	proto/frequenz/api/common/v1/microgrid/electrical_components/electrical_components.proto
